.changeset/fresh-rules-lick.md

```diff
@@ -0,0 +1,5 @@
+---
+'@itwin/itwinui-react': patch
+---
+
+Fixed internal usage of `ResizeObserver` to prevent throwing a "ResizeObserver loop limit exceeded" error in React 17 apps. Some affected components: `ButtonGroup`, `Tabs`, `Table`, `Tree`, `Breadcrumbs`, etc.
```

Comment:
Slight rephrase:

```suggestion
Updated internal usage of `ResizeObserver` to prevent throwing a benign "ResizeObserver loop limit exceeded" error in React 17 apps. Some affected components: `ButtonGroup`, `Table`, `Tree`, `Breadcrumbs`, etc.
```

because:
1. This is not "fix"ing our usage (our code was fine), it's adding a workaround. This distinction is important because we will not be porting it to v2 (which we would have done if it was an actual bug fix).
2. `Tabs` is not affected as it doesn't use the `useOverflow` hook. The prop is actually a noop.

Reply:
> 1. This is not "fix"ing our usage (our code was fine), it's adding a workaround.

Sounds good, done.

> 2\. `Tabs` is not affected as it doesn't use the `useOverflow` hook. The prop is actually a noop.

Tabs uses the `useContainerWidth` hook:

https://github.com/iTwin/iTwinUI/blob/b95b8d8deabcc98a48e4bb547e66c3a07bfeeda8/packages/itwinui-react/src/core/Tabs/Tabs.tsx#L152

… that in turn uses the `useResizeObserver` hook:

https://github.com/iTwin/iTwinUI/blob/b95b8d8deabcc98a48e4bb547e66c3a07bfeeda8/packages/itwinui-react/src/core/utils/hooks/useContainerWidth.tsx#L41

This is why I had kept in Tabs.

---

[`be5bdff` (#1800)](https://github.com/iTwin/iTwinUI/pull/1800/commits/be5bdff5794c9391751b0b6eb46fe8f3871de57f) includes the suggestion except the deletion of `Tabs`.